internal/data/receivers_wallet.go

```diff
@@ -224,34 +224,58 @@ func (rw *ReceiverWalletModel) GetByReceiverIDsAndWalletID(ctx context.Context,
 	return receiverWallets, nil
 }
 
-func (rw *ReceiverWalletModel) GetAllPendingRegistrationByReceiverWalletIDs(ctx context.Context, receiverWalletIDs []string) ([]*ReceiverWallet, error) {
-	const query = `
-		SELECT
-			rw.id,
-			rw.invitation_sent_at,
-			r.id AS "receiver.id",
-			r.phone_number AS "receiver.phone_number",
-			r.email AS "receiver.email",
-			w.id AS "wallet.id",
-			w.name AS "wallet.name"
-		FROM
-			receiver_wallets rw
-			INNER JOIN receivers r ON r.id = rw.receiver_id
-			INNER JOIN wallets w ON w.id = rw.wallet_id
-		WHERE
-			rw.id = ANY($1)
-			AND rw.status = 'READY'
-	`
+const getPendingRegistrationReceiverWalletsBaseQuery = `
+	SELECT
+		rw.id,
+		rw.invitation_sent_at,
+		r.id AS "receiver.id",
+		r.phone_number AS "receiver.phone_number",
+		r.email AS "receiver.email",
+		w.id AS "wallet.id",
+		w.name AS "wallet.name"
+	FROM
+		receiver_wallets rw
+		INNER JOIN receivers r ON r.id = rw.receiver_id
+		INNER JOIN wallets w ON w.id = rw.wallet_id
+		INNER JOIN disbursements d ON w.id = d.wallet_id
+		INNER JOIN payments p ON d.id = p.disbursement_id AND p.receiver_id = r.id
+	WHERE
+		rw.status = $1 -- 'READY'::receiver_wallet_status
+		%s
+	GROUP BY
+		rw.id,
+		r.id,
+		w.id
```

Comment:
Isn't it enough to group by receiver_wallet.id?

Reply:
It isn't. Postgres requires all these fields on the group by.